drivers/digital-io/max22200/max22200.c

```diff
@@ -0,0 +1,958 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include "max22200.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+#include "no_os_delay.h"
+
+/**
+ * @brief Read data from desired register for MAX22200
+ * @param desc - MAX22200 device descriptor.
+ * @param reg - Register value to read data from.
+ * @param val - Value read from the register.
+ * @param one_byte - true - Only the register's LSB will be returned through val
+ * 		     false - Whole 32bit register will be returned through val.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_reg_read(struct max22200_desc *desc, uint32_t reg, uint32_t *val,
+		      bool one_byte)
+{
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->buff,
+		.rx_buff = desc->buff,
+		.bytes_number = 1,
+		.cs_change = 1,
+	};
+	int ret;
+
+	desc->buff[0] = no_os_field_prep(MAX22200_RW_MASK, 0) |
+			no_os_field_prep(MAX22200_ADDR_MASK, reg) |
+			no_os_field_prep(MAX22200_ONE_BYTE_MASK, one_byte);
+
+	ret = no_os_gpio_set_value(desc->cmd_desc, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	if (!one_byte)
+		xfer.bytes_number = MAX22200_FRAME_SIZE;
+
+	ret = no_os_gpio_set_value(desc->cmd_desc, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	if (one_byte) {
```

Comment:
nit: can we do here an if else and return only once? In this manner we avoid duplicated `return 0`

Reply:
Sure.